ARCs/arc-draft_get_refs.md

```diff
@@ -0,0 +1,60 @@
+---
+arc: <to be assigned>
+title: Method Reference Discovery
+description: A standardized way contracts can reveal which references will be needed to call a specific method
+author: Joe Polny (@joe-p)
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category: ARC
+created: 2023-07-28
+requires (*optional): 4
+--- 
+
+## Abstract
+A contract caller needs to know which resources a contract needs to be availible before calling a method. This ARC proposes a standard way to make that information discoverable.
+
+## Motivation
+As of the time of this ARC, it can be hard to know which resources a caller needs to make availible when calling an application. The current solution typically involves proprietary SDKs which can make app usage and composability difficult.
+
+### Simulation Consideration
+
+It should be noted that it is currently planned for the algod simulate endpoint to allow readonly execution of a method without providing references. The response to this endpoint will return the necessary resources. Once this functionality is availible, this ARC will no longer be needed. The primary intent of this ARC is to serve as an intermediate solution.
+
+## Specification
+If an application has a method and the contract wants to make the required resources for calling the method discoverable, it **MUST** implement a readonly method with the same exact signature with an `arcXXXX_` prefix and a return type of `(address[],uint64[],uint64[],(uint64,byte[]))`. 
+
+The return value corresponds to arrays containing the required account, application, asset, and box references respectively.
```

Comment:
It would be nice to provide clarity in the case of a method which takes reference type arguments.

I.e. does its `arcXXXX_` function need to return the already specified reference types, or can it omit them?